crates/api/api_crud/src/comment/create.rs

```diff
@@ -67,21 +67,22 @@ pub async fn create_comment(
   check_community_user_action(&local_user_view, &post_view.community, &mut context.pool()).await?;
   check_post_deleted_or_removed(&post)?;
 
-  // Check if post is locked, no new comments
-  let is_mod_or_admin = is_mod_or_admin(&mut context.pool(), &local_user_view, community_id)
-    .await
-    .is_ok();
-  if post.locked && !is_mod_or_admin {
-    Err(LemmyErrorType::Locked)?
-  }
-
   // Fetch the parent, if it exists
   let parent_opt = if let Some(parent_id) = data.parent_id {
     Comment::read(&mut context.pool(), parent_id).await.ok()
   } else {
     None
   };
 
+  // Check if post or parent is locked, no new comments
+  let is_mod_or_admin = is_mod_or_admin(&mut context.pool(), &local_user_view, community_id)
+    .await
+    .is_ok();
+  let locked = post.locked || parent_opt.as_ref().is_some_and(|p| p.locked);
+  if locked && !is_mod_or_admin {
+    Err(LemmyErrorType::Locked)?
+  }
```

Comment:
I spose this should work fine, because the lockComment action locks all the children as well. Might be good to add a comment about that.